Code/Framework/AzCore/AzCore/Debug/Trace.h

```diff
@@ -39,14 +37,122 @@ namespace AZ
             None
         };
 
-        class Trace
+        class ITrace
         {
         public:
-            static Trace& Instance()    { return g_tracer; }
+            O3DEKERNEL_API ITrace()
+            {
+                tracer = this;
+            }
+            O3DEKERNEL_API virtual ~ITrace()
+            {
+                tracer = nullptr;
+            }
```

Comment:
Static variables must be start with s_lowerCamelCase per the [Coding Standard](https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#variable-names)